src/main/java/org/prebid/server/bidder/gumgum/GumgumBidder.java

```diff
@@ -138,26 +154,47 @@ private static Site modifySite(Site site, String trackingId) {
     public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
         try {
             final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
-            return Result.of(extractBids(bidResponse), Collections.emptyList());
+            return Result.of(extractBids(bidResponse, bidRequest), Collections.emptyList());
         } catch (DecodeException | PreBidException e) {
             return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
         }
     }
 
-    private static List<BidderBid> extractBids(BidResponse bidResponse) {
+    private static List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidRequest) {
         return bidResponse == null || bidResponse.getSeatbid() == null
                 ? Collections.emptyList()
-                : bidsFromResponse(bidResponse);
+                : bidsFromResponse(bidResponse, bidRequest);
     }
 
-    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse) {
+    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse, BidRequest bidRequest) {
         return bidResponse.getSeatbid().stream()
                 .map(SeatBid::getBid)
                 .flatMap(Collection::stream)
-                .map(bid -> BidderBid.of(bid, BidType.banner, DEFAULT_BID_CURRENCY))
+                .map(bid -> toBidderBid(bid, bidRequest, bidResponse.getCur()))
                 .collect(Collectors.toList());
     }
 
+    private static BidderBid toBidderBid(Bid bid, BidRequest bidRequest, String currency) {
+        final BidType bidType = getMediaType(bid.getImpid(), bidRequest.getImp());
+        final Bid.BidBuilder bidBuilder = bid.toBuilder();
```

Comment:
Seems we need builder for video.
Usually, creating of builder is expensive operation for performance, pls avoid if possible.